connectivity/tests/encryption.go

```diff
@@ -224,6 +254,125 @@ func testNoTrafficLeak(ctx context.Context, t *check.Test, s check.Scenario,
 	}
 }
 
+func testNoTrafficLeakStrict(ctx context.Context, t *check.Test,
+	client, server, clientHost *check.Pod, reqType requestType, ipFam check.IPFamily,
+) {
+	deleteCES := func(endpointName string) {
+		cesList, err := clientHost.K8sClient.CiliumClientset.CiliumV2alpha1().CiliumEndpointSlices().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Failed to list CiliumEndpointSlices: %s", err)
+		}
+		var cesToDelete string
+		for _, ces := range cesList.Items {
+			for _, ep := range ces.Endpoints {
+				if ep.Name == endpointName {
+					cesToDelete = ces.Name
+					break
+				}
+			}
+		}
+		if cesToDelete == "" {
+			t.Fatalf("Failed to find CiliumEndpointSlice for pod %s", server.Pod.Name)
+		}
+		if err := clientHost.K8sClient.CiliumClientset.CiliumV2alpha1().CiliumEndpointSlices().Delete(ctx, cesToDelete, metav1.DeleteOptions{}); err != nil {
+			t.Fatalf("Failed to delete CiliumEndpointSlice %s: %s", cesToDelete, err)
+		}
+	}
+	setCiliumOperatorScale := func(replicas int32) int32 {
+		scale, err := clientHost.K8sClient.Clientset.AppsV1().Deployments(t.Context().Params().CiliumNamespace).GetScale(ctx, "cilium-operator", metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Failed to get cilium-operator scale: %s", err)
+		}
+		savedReplicas := scale.Spec.Replicas
+		scale.Spec.Replicas = replicas
+		if _, err := clientHost.K8sClient.Clientset.AppsV1().Deployments(t.Context().Params().CiliumNamespace).UpdateScale(ctx, "cilium-operator", scale, metav1.UpdateOptions{}); err != nil {
+			t.Fatalf("Failed to scale cilium-operator: %s", err)
+		}
+		return savedReplicas
+	}
+	waitForIPCacheEntry := func(clientPod, dstPod *check.Pod) {
+		timeout := time.After(20 * time.Second)
```

Comment:
nit: Perhaps using consts here for the timeout and sleep durations would make it easier to adjust in the future?